doc/generic_callback_wrapper.md

```diff
@@ -0,0 +1,77 @@
+# GenericCallbackWrapper
+
+`Napi::GenericCallbackWrapper<T>` is a class with one template parameter `T`, 
+which is intended to allow you to return `T` from a C++ function running arbitrary async i.e. concurrency 
+regimes like `boost::asio` via a `Napi::Promise` i.e. JS promise, including exception and conversion function handling. 
+
+`Napi::GenericCallbackWrapper<T>` requires C++ exceptions enabled, so works by default.
```

Comment:
I think this should be expanded, with a reference to https://github.com/nodejs/node-addon-api/blob/fcf173d2a177cca2cc5a722117fe374513d7d1fb/doc/error_handling.md so that its clear this feature is only supported when you use the "Handling Errors With C++ Exceptions" approach.